Fixes #20: Pass access_token on after uploads.post()

TimoSta opened the pull request:
The access_token was not passed on to subsequent calls of `_check()` after
a file upload via `uploads.post()`. This caused authorization issues if
the token was not maintained in the central strava_config file but passed
in explicitly.

UnbounDev commented on test/uploads.js:
```diff
@@ -6,37 +6,36 @@
-                //done();
-            });
+        it('should upload a GPX file', function(done) {
+            new Promise((resolve, reject) => {
```
This breaks the tests on node `v0.12.x` due to the use of arrow functions! We'd have to add babel support to the repo to support these changes in a non breaking way

```
Running "simplemocha:all" (simplemocha) task
Warning: /var/www/strava/test/uploads.js:12
            new Promise((resolve, reject) => {
                                          ^^
Unexpected token => Used --force, continuing.
```

TimoSta commented:
I think the better choice would be to not use ES6 features that require babel or polyfills to run with 0.12.x to keep the library as lean as possible.
If you agree, I'm more than happy to replace the arrow functions.

As far as I know, 0.12 is in maintenance until sometime late 2016. After that, it may be a good idea to drop support for this node version with a new major version of the library and refactor the library to use more ES6 syntax. This would definitely make the code more readable and concise.

UnbounDev commented:
I'm all for that - when the respective branches have been edited to pre ES6 compliance I'll continue w/ the merge+publish

TimoSta commented:
Replaced the arrow functions with regular functions. The unit tests are now running fine all the way down to v0.10.x.

Also added a Promise polyfill to allow using promises in the unit tests without breaking compatibility with older node versions.
